artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/RefsOperation.java

```diff
@@ -190,9 +190,7 @@ private void decrementRefCount(MessageReference refmsg) {
 
    @Override
    public synchronized List<MessageReference> getRelatedMessageReferences() {
-      List<MessageReference> listRet = new LinkedList<>();
```

Comment:
this is dangerous... The RefsOperation will clear the list upon completion.
The callers could be using the list for other factors...

You really need the isolation here.


Can you close this PR? This is meant to be.